src/Catalog/Helpers/Db2CatalogSqlQueries.cs

```diff
@@ -0,0 +1,122 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Data.SqlClient;
+using System.Data.SqlTypes;
+using System.Threading.Tasks;
+
+namespace NuGet.Services.Metadata.Catalog.Helpers
+{
+    /// <summary>
+    /// Utility class for all SQL queries invoked by Db2Catalog.
+    /// </summary>
+    public class Db2CatalogSqlQueries : IDb2CatalogQueries
+    {
+        private const string CursorParameterName = "Cursor";
+        private readonly Db2CatalogProjection _db2catalogProjection;
+        private readonly int _commandTimeout;
+
+        public Db2CatalogSqlQueries(
+            PackageContentUriBuilder packageContentUriBuilder,
+            int commandTimeout)
+        {
+            _db2catalogProjection = new Db2CatalogProjection(packageContentUriBuilder);
+            _commandTimeout = commandTimeout;
+        }
+
+        public async Task<IReadOnlyCollection<DateTime>> GetTimestampsAsync(
+            SqlConnection sqlConnection,
+            SqlTransaction sqlTransaction,
+            int top,
+            DateTime since,
+            string columnName)
```

Comment:
What would be the behavior for random columnNames? Should the column names be restricted?

Reply:
Restricted in refactoring. See https://github.com/NuGet/NuGet.Services.Metadata/pull/524/commits/6ff842b66eeacec3f6e19a0dd57a5936bc911173